```
building path(s) ‘/nix/store/iacv2xfysaw35m1vkr0fnwc01dlpda0d-blackmagic-1.6.1’
unpacking sources
unpacking source archive /nix/store/zrz1qhakcin1qw6ss8glr5g6zk9i659k-blackmagic-d3a8f27fdbf952194e8fc5ce9b2fc9bcef7c545c-src
source root is blackmagic-d3a8f27fdbf952194e8fc5ce9b2fc9bcef7c545c-src
patching sources
configuring
no configure script, doing nothing
building
make: Entering directory '/build/blackmagic-d3a8f27fdbf952194e8fc5ce9b2fc9bcef7c545c-src/libopencm3'
  GENHDR  include/libopencm3/stm32/l1/irq.json
/bin/sh: ./scripts/irq2nvic_h: /usr/bin/env: bad interpreter: No such file or directory
make: *** [Makefile:56: include/libopencm3/stm32/l1/irq.json.genhdr] Error 126
make: Leaving directory '/build/blackmagic-d3a8f27fdbf952194e8fc5ce9b2fc9bcef7c545c-src/libopencm3'

builder for ‘/nix/store/jbrsd8s5lrribi01hg1vp3v12h8lz6h2-blackmagic-1.6.1.drv’ failed with exit code 2
error: build of ‘/nix/store/jbrsd8s5lrribi01hg1vp3v12h8lz6h2-blackmagic-1.6.1.drv’ failed

```

--- a/pkgs/development/tools/misc/blackmagic/default.nix
+++ b/pkgs/development/tools/misc/blackmagic/default.nix
@@ -1,0 +1,59 @@
+{ stdenv, lib, fetchFromGitHub
+, gcc-arm-embedded, bash, libftdi
+, python, pythonPackages
+}:
+
+with lib;
+
+stdenv.mkDerivation rec {
+  name = "blackmagic-${version}";
+  version = "1.6.1";
+
+  src = fetchFromGitHub {
+    owner = "blacksphere";
+    repo = "blackmagic";
+    rev = "d3a8f27fdbf952194e8fc5ce9b2fc9bcef7c545c";
+    sha256 = "0c3l7cfqag3g7zrfn4mmikkx7076hb1r856ybhhdh0f6zji2j6jx";
+    fetchSubmodules = true;
+  };
+
+  buildInputs = [
+    gcc-arm-embedded
+    libftdi
+    python
+    pythonPackages.intelhex
+  ];
+
+  postPatch = ''
+    # Prevent calling out to `git' to generate a version number:
+    substituteInPlace src/Makefile \
+      --replace '`git describe --always --dirty`' '${version}'
+
+    # Fix scripts that generate headers:
+    for f in $(find scripts libopencm3/scripts -type f); do
+      patchShebangs "$f"
+    done
+  '';
+
+  buildPhase = "${stdenv.shell} ${./helper.sh}";
+  installPhase = ":"; # buildPhase does this.
+
+  meta = {
+    description = "In-application debugger for ARM Cortex microcontrollers";
+    longDescription = ''
+      The Black Magic Probe is a modern, in-application debugging tool
+      for embedded microprocessors. It allows you to see what is going
+      on "inside" an application running on an embedded microprocessor
+      while it executes.
+
+      This package builds the firmware for all supported platforms,
+      placing them in separate directories under the firmware
+      directory.  It also places the FTDI version of the blackmagic
+      executable in the bin directory.
+    '';
+    homepage = https://github.com/blacksphere/blackmagic;
+    license = licenses.gpl3Plus;
+    maintainers = with maintainers; [ pjones ];
+    platforms = platforms.unix;
+  };
+}

--- a/pkgs/development/tools/misc/blackmagic/helper.sh
+++ b/pkgs/development/tools/misc/blackmagic/helper.sh
@@ -1,0 +1,49 @@
+################################################################################
+# Build all of the platforms manually since the `all_platforms' target
+# doesn't preserve all of the build outputs and overrides CFLAGS.
+set -e
+set -u
+
+################################################################################
+# Prevent a warning from shellcheck:
+out=${out:-/tmp}
+
+################################################################################
+export CFLAGS=$NIX_CFLAGS_COMPILE
+
+################################################################################
+PRODUCTS="blackmagic.bin blackmagic.hex blackmagic_dfu.bin blackmagic_dfu.hex"
+
+################################################################################
+make_platform() {
+  echo "Building for hardware platform $1"
+
+  make clean
+  make PROBE_HOST="$1"
+
+  if [ "$1" = libftdi ]; then
+    mkdir -p "$out/bin"
+    install -m 0555 blackmagic "$out/bin"
+  fi
+
+  for f in $PRODUCTS; do
+    if [ -r "$f" ]; then
+      mkdir -p "$out/firmware/$1"
+      install -m 0444 "$f" "$out/firmware/$1"
+    fi
+  done
+
+}
+
+################################################################################
+# Start by building libopencm3:
+make -C libopencm3
+
+################################################################################
+# And now all of the platforms:
+cd src
+
+for platform in platforms/*/Makefile.inc; do
+  probe=$(basename "$(dirname "$platform")")
+  make_platform "$probe"
+done

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -6500,6 +6500,8 @@ with pkgs;
   bison3 = callPackage ../development/tools/parsing/bison/3.x.nix { };
   bison = bison3;
 
+  blackmagic = callPackage ../development/tools/misc/blackmagic { };
+
   bloaty = callPackage ../development/tools/bloaty { };
 
   bossa = callPackage ../development/tools/misc/bossa {

